File: library/src/main/java/org/meganekkovr/animation/ScaleUpdateListener.java
package org.meganekkovr.animation;

import android.animation.ValueAnimator;
import android.support.annotation.NonNull;

import org.joml.Vector3f;
import org.meganekkovr.Entity;

class ScaleUpdateListener extends TransformUpdateListener {

    public ScaleUpdateListener(@NonNull Entity target) {
        super(target);
    }

    @Override
    public void onAnimationUpdate(@NonNull ValueAnimator animation) {
        target.setScale((Vector3f) animation.getAnimatedValue());
    }
}

File: library/src/main/java/org/meganekkovr/animation/ScaleUpdateListener.kt
package org.meganekkovr.animation

import android.animation.ValueAnimator

import org.joml.Vector3f
import org.meganekkovr.Entity

internal class ScaleUpdateListener(target: Entity) : TransformUpdateListener(target) {

    override fun onAnimationUpdate(animation: ValueAnimator) {
        target.scale = animation.animatedValue as Vector3f
    }
}
